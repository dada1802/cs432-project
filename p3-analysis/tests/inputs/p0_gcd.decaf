def int gcd1(int a, int b)
{
    int temp;
    while (b != 0) {
        temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

def int gcd2(int a, int b)
{
    while (a != b)
    {
        if (a > b)
        {
            a = a - b;
        }
        else
        {
            b = b - a;
        }
    }
    return a;
}

def int gcd3(int a, int b)
{
        int result;
        if(a > b)
        {
            result = b;
        }
        else
        {
            result = a;
        }
        while (result > 0) {
            if (a % result == 0 && b % result == 0) {
                break;
            }
            result = result - 1;
        }

        // Return gcd of a and b
        return result;
}

def int main()
{
    return 0;
}