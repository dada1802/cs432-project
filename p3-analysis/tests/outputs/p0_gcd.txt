Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 min : (int, int) -> int
 gcd : (int, int) -> int
 main : () -> int

  FuncDecl name="min" return_type=int parameters={a:int,b:int} [line 1]
  SYM TABLE:
   a : int
   b : int

    Block [line 2]
    SYM TABLE:

        Block [line 4]
        SYM TABLE:

  FuncDecl name="gcd" return_type=int parameters={a:int,b:int} [line 11]
  SYM TABLE:
   a : int
   b : int

    Block [line 12]
    SYM TABLE:
     m : int

        Block [line 17]
        SYM TABLE:

            Block [line 19]
            SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 27]
  SYM TABLE:

    Block [line 28]
    SYM TABLE:
     a : int

