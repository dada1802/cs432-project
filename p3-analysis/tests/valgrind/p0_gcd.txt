==134214== Memcheck, a memory error detector
==134214== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==134214== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==134214== Command: ../decaf inputs/p0_gcd.decaf
==134214== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 gcd1 : (int, int) -> int
 gcd2 : (int, int) -> int
 gcd3 : (int, int) -> int
 main : () -> int

  FuncDecl name="gcd1" return_type=int parameters={a:int,b:int} [line 1]
  SYM TABLE:
   a : int
   b : int

    Block [line 2]
    SYM TABLE:
     temp : int

        Block [line 4]
        SYM TABLE:

  FuncDecl name="gcd2" return_type=int parameters={a:int,b:int} [line 12]
  SYM TABLE:
   a : int
   b : int

    Block [line 13]
    SYM TABLE:

        Block [line 15]
        SYM TABLE:

            Block [line 17]
            SYM TABLE:

            Block [line 21]
            SYM TABLE:

  FuncDecl name="gcd3" return_type=int parameters={a:int,b:int} [line 28]
  SYM TABLE:
   a : int
   b : int

    Block [line 29]
    SYM TABLE:
     result : int

        Block [line 32]
        SYM TABLE:

        Block [line 36]
        SYM TABLE:

        Block [line 39]
        SYM TABLE:

            Block [line 40]
            SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 50]
  SYM TABLE:

    Block [line 51]
    SYM TABLE:

==134214== 
==134214== HEAP SUMMARY:
==134214==     in use at exit: 0 bytes in 0 blocks
==134214==   total heap usage: 2,645 allocs, 2,645 frees, 347,536 bytes allocated
==134214== 
==134214== All heap blocks were freed -- no leaks are possible
==134214== 
==134214== For lists of detected and suppressed errors, rerun with: -s
==134214== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
