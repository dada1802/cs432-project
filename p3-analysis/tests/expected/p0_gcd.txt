Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 gcd1 : (int, int) -> int
 gcd2 : (int, int) -> int
 gcd3 : (int, int) -> int
 main : () -> int

  FuncDecl name="gcd1" return_type=int parameters={a:int,b:int} [line 1]
  SYM TABLE:
   a : int
   b : int

    Block [line 2]
    SYM TABLE:
     temp : int

        Block [line 4]
        SYM TABLE:

  FuncDecl name="gcd2" return_type=int parameters={a:int,b:int} [line 12]
  SYM TABLE:
   a : int
   b : int

    Block [line 13]
    SYM TABLE:

        Block [line 15]
        SYM TABLE:

            Block [line 17]
            SYM TABLE:

            Block [line 21]
            SYM TABLE:

  FuncDecl name="gcd3" return_type=int parameters={a:int,b:int} [line 28]
  SYM TABLE:
   a : int
   b : int

    Block [line 29]
    SYM TABLE:
     result : int

        Block [line 32]
        SYM TABLE:

        Block [line 36]
        SYM TABLE:

        Block [line 39]
        SYM TABLE:

            Block [line 40]
            SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 50]
  SYM TABLE:

    Block [line 51]
    SYM TABLE:

